[Bug] get status failed

**Bug Description**
E   AttributeError: 'SandboxManager' object has no attribute 'logger'

**Steps to Reproduce**

**Expected Behavior**

**Actual Behavior**

**Error Logs**

**Environment Information**
- **OS**: [e.g. Ubuntu 22.04, macOS 13.0, Windows 11]
- **Python Version**: [e.g. 3.11.5]
- **ROCK Version**: [e.g. 0.2.0]
- **Installation Method**: [e.g. pip install rl-rock, source installation with uv]
- **Docker Version**: [e.g. 24.0.6] (if using sandbox features)
- **Deployment Type**: [e.g. local, distributed, ray]

**ROCK Configuration**
- **Runtime Environment Type**: [e.g. uv, pip, conda]
- **Sandbox Image**: [e.g. python:3.11, custom image]
- **Resource Allocation**: [e.g. memory=8g, cpus=2.0]

**Component Affected**
- [x] Sandbox
- [ ] Actions
- [ ] Deployments
- [ ] SDK & API
- [ ] Envhub
- [ ] CLI 
- [ ] Performance & Optimization
- [ ] Documentation & Examples

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -106,6 +106,7 @@ test = [
     "pytest-trio",
     "pytest-twisted",
     "pytest-env",
+    "fakeredis[json]",
 ]
 
 [tool.setuptools.packages.find]

--- a/rock/sandbox/sandbox_manager.py
+++ b/rock/sandbox/sandbox_manager.py
@@ -33,7 +33,6 @@ from rock.rocklet import __version__ as swe_version
 from rock.sandbox import __version__ as gateway_version
 from rock.sandbox.base_manager import BaseManager
 from rock.sandbox.sandbox_actor import SandboxActor
-from rock.utils import SANDBOX_ID
 from rock.utils.providers import RedisProvider
 
 logger = init_logger(__name__)
@@ -215,7 +214,7 @@ class SandboxManager(BaseManager):
                 sandbox_info.update(remote_status.to_dict())
                 await self._redis_provider.json_set(alive_sandbox_key(sandbox_id), "$", sandbox_info)
                 await self._update_expire_time(sandbox_id)
-                self.logger.info(f"sandbox {sandbox_id} status is {remote_status}, write to redis")
+                logger.info(f"sandbox {sandbox_id} status is {remote_status}, write to redis")
             else:
                 sandbox_info = await self.async_ray_get(sandbox_actor.sandbox_info.remote())
 

--- a/tests/unit/sandbox/test_sandbox_manager.py
+++ b/tests/unit/sandbox/test_sandbox_manager.py
@@ -3,6 +3,7 @@ import time
 
 import pytest
 import ray
+from fakeredis import aioredis
 
 from rock.actions import SandboxStatusResponse
 from rock.config import RockConfig
@@ -10,13 +11,22 @@ from rock.deployments.config import DockerDeploymentConfig, RayDeploymentConfig
 from rock.deployments.constants import Port
 from rock.deployments.status import ServiceStatus
 from rock.sandbox.sandbox_manager import SandboxManager
+from rock.utils.providers.redis_provider import RedisProvider
 
 
 @pytest.fixture
-async def sandbox_manager(rock_config: RockConfig, ray_init_shutdown):
+async def redis_provider():
+    provider = RedisProvider(host=None, port=None, password="")
+    provider.client = aioredis.FakeRedis(decode_responses=True)
+    yield provider
+    await provider.close_pool()
+
+
+@pytest.fixture
+async def sandbox_manager(rock_config: RockConfig, redis_provider: RedisProvider, ray_init_shutdown):
     sandbox_manager = SandboxManager(
         rock_config,
-        redis_provider=None,
+        redis_provider=redis_provider,
         ray_namespace=rock_config.ray.namespace,
         enable_runtime_auto_clear=rock_config.runtime.enable_auto_clear,
     )

